api/src/server.ts

```diff
@@ -155,7 +155,7 @@ export async function startServer(): Promise<void> {
 	const server = await createServer();
 
 	const host = env['HOST'] as string;
-	const port = parseInt(env['PORT'] as string);
+	const port = env['PORT'];
 
 	server
 		.listen(port, host, () => {
```

Comment:
As far as I can tell, `express` doesn't accept a `host` parameter when a socket is used instead of a port. We'll have to update this to drop the host if a socket is passed instead of a port. 

Also, is it kosher to put a socket path inside the env var named `PORT`, or should we support a new `SOCKET`? I'm not familiar with the standards/best practices around this, but it feels odd to have a socket path given in a port 🤔

Reply:
Yea, I also recognized that tests are failing because of this :(

And yes, it is best practice to use `PORT`. I don't know what would happened if we use `SOCKET` and `PORT` is already confirmed working :-)

Reply:
I say we add a new `PATH` variable which accepts a socket. This paradigm is used by [express](https://expressjs.com/en/5x/api.html#app.listen_path_callback), [node](https://nodejs.org/api/net.html#serverlisten) and other popular web frameworks (e.g. `Koa`, `fastify` etc) for listening to a socket connection. That is as about as much of a "standard" as we will get. Thoughts?

Reply:
Seems like you are a way better expert in this area. Sound like a good option especially if other frameworks already use it like that

Reply:
@ComfortablyCoding  Sadly I'm not sure how to implent this in the most performant way. When I look at the documentation of express that you linked it seems like we need to init the listen process in another step. To get rid of the host parameter within the listen function in the current state:

`server
		.listen(port, host, () => {`

In my limited understanding I would add a simple if condition to check if the path variable is set. 

And if it is set I would execute another branch that looks like this

`server
		.listen(path, () => {`

I don't know if there is a more elegant way to do this

Reply:
You should be able to use the `server.listen(options)` overload[^1] and throw in the host, port and path altogether and let listen figure it out. (ignoring `path` if `port` is specified).

I also advocate to not call the env variable `PATH`, which conflicts with the actual `PATH` env variable, but rather something like `SERVER_PATH`?

[^1]: https://nodejs.org/api/net.html#serverlistenoptions-callback

Reply:
@McSundae 

The `server.listen` that we use is actually the node http server version. Due to this we could use the [options](https://nodejs.org/api/net.html#serverlistenoptions-callback) signature that supports all these params.

Reply:
> I also advocate to not call the env variable `PATH`, which conflicts with the actual `PATH` env variable, but rather something like `SERVER_PATH`?

`UNIX_PATH` or `UNIX_SOCKET_PATCH` might be more intuitive

Reply:
I think everything is fine now